### DISCUSSION TITLE
Transparency + Colors lost after Resize of a Transparent Image

### DISCUSSION THREAD

**Discussion by Maxxxel** [SixLabors/ImageSharp#2222] [Q&A] [answered]
```csharp
PngEncoder encoder = new PngEncoder();
encoder.ColorType = PngColorType.RgbWithAlpha;
encoder.TransparentColorMode = PngTransparentColorMode.Preserve;
encoder.BitDepth = PngBitDepth.Bit8;
encoder.CompressionLevel = PngCompressionLevel.BestSpeed;

using (Image<SixLabors.ImageSharp.PixelFormats.Rgba32> image = Image.Load<SixLabors.ImageSharp.PixelFormats.Rgba32>(stream))
{
  ResizeOptions resizeOptions = new ResizeOptions
  {
    Mode = ResizeMode.Max,
    PremultiplyAlpha = true
  };
  image.Mutate(x => x.Resize(Math.Min(image.Width, 1024), Math.Min(image.Height, 1024))); 
  image.SaveAsPngAsync(Exporter.Folder + "\\" + textureNames[3] + ".png", encoder);
}
```

Like the Titel says, when I run this on an Image with Alpha = 0, the resized image becomes a totally blank one where all values RGBA are zero.

Image i try that for: 
![material_1_baseColor](https://user-images.githubusercontent.com/10586721/187800824-e910ba2c-c973-40e0-ae91-166fc87920f8.png)

**Comment by JimBobSquarePants** [comment, marked as answer]
I don't fully understand what you are saying here. Are you saying that if you resize an image where the alpha component of the image is 0, then the result has 0 for red, green, blue components?

If so, that makes perfect sense since you are pre-multiplying the other components by the alpha value. You should set `ResizeOptions.PremultiplyAlpha` to `false` to avoid this.

**Reply by Maxxxel** [reply]
I tried it, still all values turn Into zeros....

**Reply by JimBobSquarePants** [reply]
It shouldn't. Do you have an input image?

**Reply by Maxxxel** [reply]
The one I provided at first post

**Reply by JimBobSquarePants** [reply]
Sorry, that wasn't obvious. 

I cannot replicate this. You know the property defaults to `true` yeah? You need to explicitly set it to `false`.

Here's the resized output. Scaled to 50%
![Resize_NoAlpha](https://user-images.githubusercontent.com/385879/187925061-4404080d-7e69-4bb5-8b15-04da3dcf55ac.png)

**Reply by Maxxxel** [reply]
I'm at work now, I reply in 1.5hrs with my tests. Thanks.

**Reply by Maxxxel** [reply]
@JimBobSquarePants 

With the underneath code my image turns all ARGB into zero values. Where is the mistake???
```csharp
using SixLabors.ImageSharp.Formats.Png;
using SixLabors.ImageSharp.Processing;
using SixLabors.ImageSharp;

PngEncoder encoder = new PngEncoder();
encoder.ColorType = PngColorType.RgbWithAlpha;
encoder.TransparentColorMode = PngTransparentColorMode.Preserve;
encoder.BitDepth = PngBitDepth.Bit8;
encoder.CompressionLevel = PngCompressionLevel.BestSpeed;

using (Image<SixLabors.ImageSharp.PixelFormats.Rgba32> image = Image.Load<SixLabors.ImageSharp.PixelFormats.Rgba32>("test.png"))
{
    ResizeOptions resizeOptions = new ResizeOptions
    {
        Mode = ResizeMode.Max,
        PremultiplyAlpha = false
    };
    image.Mutate(x => x.Resize(Math.Min(image.Width, 1024), Math.Min(image.Height, 1024)));
    image.SaveAsPngAsync("result.png", encoder);
}
```

**Reply by Maxxxel** [reply]
OMG! 

Sorry for wasting your time... Just noticed i dont even use the resize Options....
```csharp
using SixLabors.ImageSharp.Formats.Png;
using SixLabors.ImageSharp.Processing;
using SixLabors.ImageSharp;

PngEncoder encoder = new PngEncoder
{
    ColorType = PngColorType.RgbWithAlpha,
    TransparentColorMode = PngTransparentColorMode.Preserve,
    BitDepth = PngBitDepth.Bit8,
    CompressionLevel = PngCompressionLevel.BestSpeed
};

ResizeOptions resizeOptions = new ResizeOptions
{
    Mode = ResizeMode.Max,
    PremultiplyAlpha = false,
    Size = new Size(1024, 1024)
};

using (Image<SixLabors.ImageSharp.PixelFormats.Rgba32> image = Image.Load<SixLabors.ImageSharp.PixelFormats.Rgba32>("test.png"))
{
    image.Mutate(x => x.Resize(resizeOptions));
    image.SaveAsPng("result.png", encoder);
}
```

Everything works as expected, very nice Lib!!!!!!!!!!!

**Reply by JimBobSquarePants** [reply]
Don't worry about it. I missed it too! 😄
